sample_apps/simple_app/run_example.sh

```diff
@@ -156,6 +156,26 @@ function run_cmd() {
    set +x
 }
 
+# ###########################################################################
+# Deep-clean the build-env to remove artifacts (e.g. generated files, binaries
+# etc.) that may have been produced by other steps. We run this to ensure
+# that this script will run successfully w/o any dependencies on executing
+# some prior steps.
+# ###########################################################################
+function rm_non_git_files() {
```

Comment:
This new function will always run, to delete non-Git-owned objects.

This helps test the makefile changes in an env where nothing generated previously exists.